File: agent-driven-architecture-main/examples/typescript/src/blocks/console/contract.ts
// ── blocks/console/contract — A PRESENTATION BLOCK'S TRANSPORT (6.8) ───────
// Read this beside `blocks/triage/contract.ts`. They are the same file with
// different nouns.
//
// 6.8's "a UI tool folds, does not sign" carve-out is DELETED. An agent that can
// show/hide, reposition and restructure the interface — auditably and
// replayably — is a primary advantage of this architecture, and unsigning UI
// tools removes exactly that. `focusTicket` and `setPanel` are Command cases
// with the same shape, the same signer, the same commit path and the same blast
// radius as `setPriority`. There is no second table and no apologetic caption.
//
// The axis is 4.6's, and 4.6 is untouched: a DECISION about presentation is an
// authored act — it folds and signs. EPHEMERAL local view-state (hover, scroll
// offset, unsubmitted text) never enters a tool; it lives in `view-state.ts`.
// "Why did the escalation button disappear?" is a question a human needs
// answered, so the answer is on the bus.

import type { CommandBase } from "@adr/spine/pure/command";
import type { PanelId, TicketId } from "@adr/spine/pure/ids";
import type { ToolResultBase } from "@adr/spine/pure/tool-result";
import { claims } from "@adr/spine/pure/tool-result";

// ── ToolResult cases ────────────────────────────────────────────────────────
export interface FocusTicketResult extends ToolResultBase {
  readonly outcome: "ok";
  readonly tool: "focusTicket";
  readonly ticket: TicketId;
}

export interface SetPanelResult extends ToolResultBase {
  readonly outcome: "ok";
  readonly tool: "setPanel";
  readonly panel: PanelId;
  readonly visible: boolean;
}

export type ConsoleResult = FocusTicketResult | SetPanelResult;

// ── Command cases ───────────────────────────────────────────────────────────
export interface FocusTicketCommand extends CommandBase {
  readonly outcome: "ok";
  readonly tool: "focusTicket";
  readonly ticket: TicketId;
}

export interface SetPanelCommand extends CommandBase {
  readonly outcome: "ok";
  readonly tool: "setPanel";
  readonly panel: PanelId;
  readonly visible: boolean;
}

export type ConsoleCommand = FocusTicketCommand | SetPanelCommand;

// This block contributes NO effect case. That is a property of these particular
// verbs (a layout decision is truth, not an outside-world action), not of
// presentation blocks in general.

/** WHICH RESULTS THIS BLOCK'S ARM FOLDS, derived rather than written. The table
 *  is a mapped type over the union above (`spine/pure/tool-result`), so a case
 *  added there and a case claimed here are ONE edit: omit it and the property is
 *  missing, name a tool the union does not declare and the property is excess.
 *  The under-claiming predicate this used to be — green build, stale clause,
 *  fall-through at run time — is not writable in this form. */
export const isConsoleResult = claims<ConsoleResult>({ focusTicket: true, setPanel: true });

File: agent-driven-architecture-main/examples/kotlin/spine/src/main/kotlin/adr/blocks/console/Contract.kt
// ── blocks/console/contract — A PRESENTATION BLOCK, and it SIGNS (6.8) ─────
// 6.8's "a UI tool folds, does not sign" carve-out is deleted. An agent that can
// show/hide, reposition and restructure the interface — AUDITABLY AND REPLAYABLY —
// is a primary advantage of this architecture, and unsigning UI tools removes
// exactly that.
//
// FocusTicket and SetPanel are Command cases with the same shape, the same signer,
// the same commit path and the same blast radius as SetPriority. There is no second
// table, no "folds, does not sign" row, and no apologetic caption. 3.2 ("a person
// tapping a control and the agent calling a tool resolve to the identical Command")
// and 4.4 ("the authoring discipline is identical") become true as written.
//
// The axis is 4.6's, untouched: a DECISION about presentation is truth and folds;
// EPHEMERAL local view-state (hover, scroll, unsubmitted text) never enters a tool.
// That line lives in blocks/console/view-state, one file away.

package adr.contract

import adr.spine.pure.CommandId
import adr.spine.pure.PanelId
import adr.spine.pure.Signature
import adr.spine.pure.TicketId
import adr.spine.pure.ToolName

/**
 * A sealed CLASS extending the sealed CLASS ToolResult: `tool` is passed up the chain,
 * so every variant carries it by construction rather than by re-implementing it.
 */
sealed class ConsoleResult(override val tool: ToolName) : ToolResult(tool) {
    data class FocusTicket(
        override val tool: ToolName,
        val ticket: TicketId,
    ) : ConsoleResult(tool)

    data class SetPanel(
        override val tool: ToolName,
        val panel: PanelId,
        val visible: Boolean,
    ) : ConsoleResult(tool)
}

/**
 * A sealed CLASS extending the sealed CLASS Command: tool/sig/id pass up the chain and
 * every variant carries authorship, permission and identity by construction (G12).
 */
sealed class ConsoleCommand(
    override val tool: ToolName,
    override val sig: Signature,
    override val id: CommandId,
) : Command(tool, sig, id) {
    data class FocusTicket(
        override val tool: ToolName,
        override val sig: Signature,
        override val id: CommandId,
        val ticket: TicketId,
    ) : ConsoleCommand(tool, sig, id)

    data class SetPanel(
        override val tool: ToolName,
        override val sig: Signature,
        override val id: CommandId,
        val panel: PanelId,
        val visible: Boolean,
    ) : ConsoleCommand(tool, sig, id)
}
